src/main/java/com/ahsoka/SALC/user_model/persistance/entity/User.java

```diff
@@ -0,0 +1,59 @@
+package com.ahsoka.SALC.user_model.persistance.entity;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.data.elasticsearch.annotations.Document;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+
+@EqualsAndHashCode
+@NoArgsConstructor
+@Document(indexName = "users", shards = 1, createIndex = false)
+public class User implements UserDetails {
+
+    @Getter @Setter private String id;
```

Comment:
La notación @Getter @Setter se incluye una única vez, previo al public class << ClassName >>. De esta manera, indicamos a la clase, que implemente los "getters" and "setters" de los atributos non-final.

Reply:
Tal y como has hecho con ```@EqualsAndHashCode```. De esa forma, le estamos diciendo a la clase que implemente el método equals y el hashCode. (Siempre que se cambie el equals de una clase, se debe cambiar el hashCode obligatoriamente). Ref: "Effective Java - Bloch, Josua" [Item 11]